app/src/controllers/ChannelController.js

```diff
@@ -0,0 +1,224 @@
+const connext = require('@connext/client')
+
+class ChannelController {
+  /**
+   * @constructor
+   * @param {Object} opts
+   */
+  constructor(opts) {
+    this.networkController = opts.networkController
+    this.keyringController = opts.keyringController
+    this.store = opts.store
+  }
+
+  /**
+   * Helper method for initializing the connext client.
+   *
+   * This should be called any time on app startup, or before
+   * using any channel methods.
+   *
+   * @returns Promise<void>
+   */
+  initializeConnext() {
+    const { ETH_PROVIDER_URL, NODE_URL, LOG_LEVEL } = process.env
+
+    const xpub = this.keyringController.getChannelXPub()
+    const keyGen = this.keyringController.getChannelKeyGen()
+
+    const connectOpts = {
+      xpub,
+      keyGen,
+      ethProviderUrl: ETH_PROVIDER_URL || 'https://rinkeby.indra.connext.network/api/ethprovider', // default to nodes provider
+      logLevel: LOG_LEVEL || 5, // default to log everything
+      nodeUrl: NODE_URL || 'wss://rinkeby.indra.connext.network/api/messaging' // default to rinkeby
```

Comment:
please provide us the prod/mainnet urls for this.
If you can connect and bring in the networkcontroller into this and choose the url based on network, that would be ideal

Reply:
Done 👍